### PR TITLE
Add attributes mito_name and macs_gsize to fasta asset

### PR DESCRIPTION
**Pull request by mirpedrol**
Add a new asset class child from "fasta" asset class. Add attributes mito_name and macs_gsize.

### CODE REVIEW

**Comment by mirpedrol** [comment]
Hi @nsheff!
I was trying to add attributes to the fasta asset as we were discussing. 
I can't find documentation about this kind of assets, is there a way to test that before merging the PR? 
Thanks!

File: recipes/fasta_attributes_asset_recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  Attributes to Sequences in the FASTA file. 
+  mito_name (Mitocondrial chromosome name) and macs_gsize (Genome size).
+inputs:
+  files: {}
+  params:
+    name:
+      description: Mitocondrial chromosome name
+    size:
+      description: Genome size
+  assets: {}
```
Comment (nsheff) [inline recipes/fasta_attributes_asset_recipe.yaml]: Wouldn't this need to have a fasta asset as input?

File: recipes/fasta_attributes_asset_recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+container: docker.io/databio/refgenie
+default_tag: ''
+command_template_list:
```
Comment (nsheff) [inline recipes/fasta_attributes_asset_recipe.yaml:21]: The commands here should create the asset of interest. These commands aren't going to create any persistent output

**Comment by mirpedrol** [comment]
Thank you very much for your review @nsheff. 
These attributes are values that we provide by had ([example](https://github.com/nf-core/rnaseq/blob/e049f51f0214b2aef7624b9dd496a404a7c34d14/conf/igenomes.config#L35)), can I do this by creating an env variable? Or must they be obtained from a fasta file?

**Comment by nsheff** [comment]
Got it, now I remember what you're trying to do here. I think this is the way to do this:

1. for the seek keys, use `value: "{params.name}"` and `value: "{params.size}"` to access the input parameters.
2. I believe your `command_template_list` should just be empty -- those values will automatically be made available as seek keys in the `params` namespace.

That should work... I'll try to test later today.

But the other idea is for this to instead extend the fasta asset, instead of being an entirely new asset type. In that case you'd want to do it a bit differently -- these seek keys would be the same, but then we'd be adding them to a typical fasta asset, so you'd do something like `seek genome/fasta.name`

So in other words, the question is just: should this be a separate asset that uses a fasta asset as input? Or should it be a *child* of the fasta asset, such that it *is* a fasta asset with some extra capabilities?

**Comment by mirpedrol** [comment]
Ah ok I get it now, thanks for the clarification. 
Both options should work for us, but I think the second one makes more sense. I will try to add the changes 🙂 
The only potential problem I can think of now is the case where we only have one of these two variables, will this be a problem for the asset? I assume we can always set them as empty strings.
